src/number/NumberTypes.jl

```diff
@@ -323,3 +322,34 @@ mutable struct n_unknown{T <: Nemo.RingElem} <: Nemo.RingElem
    parent::CoefficientRing{T}
 
 end
+
+###############################################################################
+#
+#   N_UnknownSingularCoefficientRing
+#   singular coefficient rings with no identifiable julia object
+#
+###############################################################################
+
+mutable struct N_UnknownSingularCoefficientRing <: Ring
```

Comment:
Doesn't this duplicate the functionality of `struct CoefficientRing` together with `struct n_unknown`, defined just above? Or if not, what's the difference? (In the latter case, a comment should explain it?)

Reply:
The `CoefficientRing{T}` type has a field `base_ring::Nemo.Ring` which we don't always have. This new type is really for the coefficient rings that singular creates that we cannot identify/there is no meaningful equivalent in julia. For example, singular has floating point coefficients, and I am not really keen on creating a special floating point coefficient type for singular.jl.

The comment is a few lines up:
```
#   N_UnknownSingularCoefficientRing
#   singular coefficient rings with no identifiable julia object
```

Reply:
I know this new unknown coefficient type is not elegant, but I don't see another way if we are not able to identify what the coefficient ring is. It is meant as a default case in this switch: https://github.com/oscar-system/Singular.jl/blob/80818307578418548e113335e4392dba6578442a/src/caller.jl#L72